src/Cardano/Transaction.cpp

```diff
@@ -67,6 +67,26 @@ uint256_t TokenBundle::getAmount(const std::string& key) const {
     return findkey->second.amount;
 }
 
+set<string> TokenBundle::getPolicyIds() const {
+    set<string> policyIds;
+    for (const auto& t: bundle) {
+        if (policyIds.find(t.second.policyId) == policyIds.end()) {
+            policyIds.emplace(t.second.policyId);
+        }
+    }
```

Comment:
Here we can use `policyIds.contains(...)` instead of `find`

and we can either use `std::copy_if` or `std::set_difference`

can also use `std::unordered_set` I believe